### DISCUSSION TITLE
Usages of the Options struct

### DISCUSSION THREAD

**Discussion by NeverGivinUp** [mgeisler/textwrap#350] [Q&A] [answered]
I'm trying to understand the `Options` struct. 

The way I see it, the `width` parameter of the `Options` struct is only used by the `wrap` function. It is not used, when calling `split_words`, `word_separator.find_words`, `wrap_first_fit` or `wrap_optimal_fit` with an explicit `line_widths` function.

When calling the core functions separately, like in the WASM-example, the `Options` struct is just required to gain access to the `splitter` and the `word_separator`.

Is that correct?

**Comment by mgeisler** [comment, marked as answer]
Hey @NeverGivinUp, yes, that sounds pretty accurate.

My goal was to make the "top-level" functions easy to use for simple things, while also making it possible to customize things further when needed. So you can do

```rust
let wrapped = textwrap::wrap(some_text, 80);
```

or you can do

```rust
let wrapped = textwrap::wrap(some_text, Options::new(80).break_words(false).initial_indent("  "));
```

The `Options` struct is the centralized place which holds the configuration options. When using the `core` functions, you could just as well create a `splitter` and `word_separator` yourself -- in the WASM demo, I just used `Options::new(width)` since it gives me the default `splitter` and `word_separator` in an easy way.

If your point is that `split_words` could take a `WordSplitter` directly, then I think I agree... I don't think there's any particular reason for it to take a full `Options` object. It would probably make things clearer if `split_words` only took what it actually needs!

**Comment by NeverGivinUp** [comment]
>   It would probably make things clearer if split_words only took what it actually needs!

Yes, I was mostly confused by the WASM demo using width in several places and wanted to make sure I'm not just overlooking the use of it.

I like the simplicity of the use of the top-level function very much :)
